src/Uno.UI/UI/Xaml/Controls/TreeView/TreeView.cs

```diff
@@ -0,0 +1,358 @@
+							UpdateItemsSelectionMode(true);
+						}
+						break;
+
+				}
+			}
+			else if (property == ItemsSourceProperty)
+			{
+				m_rootNode.IsContentMode = true;
+
+				if (ListControl != null)
+				{
+					var viewModel = ListControl.ListViewModel;
+					viewModel.IsContentMode = true;
+				}
+
+				m_rootNode.ItemsSource = ItemsSource;
+			}
+		}
+
+		private void OnListControlDragItemsStarting(object sender, DragItemsStartingEventArgs args)
+		{
+			var treeViewArgs = new TreeViewDragItemsStartingEventArgs(args);
+			//TODO: m_dragItemsStartingEventSource(*this, *treeViewArgs);
+		}
+
+		private void OnListControlDragItemsCompleted(object sender, DragItemsCompletedEventArgs args)
+		{
+			//		const auto newParent = [items = args.Items(), listControl = ListControl(), rootNode = m_rootNode.get()]()
+
+
+			//	{
+			//			if (listControl && items && items.Size() > 0)
+			//			{
+			//				if (const auto draggedNode = listControl->NodeFromItem(items.GetAt(0)))
+			//            {
+			//					const auto parentNode = draggedNode.Parent();
+			//					if (parentNode && parentNode != rootNode)
+			//					{
+			//						return listControl->ItemFromNode(parentNode);
+			//					}
+			//				}
+			//			}
+			//			return static_cast<winrt::IInspectable>(nullptr);
+			//		} ();
+
+			//		const auto treeViewArgs = winrt::make_self<TreeViewDragItemsCompletedEventArgs>(args, newParent);
+			//		m_dragItemsCompletedEventSource(*this, *treeViewArgs);
+		}
+
+
+		private void UpdateItemsSelectionMode(bool isMultiSelect)
+		{
+			var listControl = ListControl;
+			listControl.EnableMultiselect(isMultiSelect);
+
+			var viewModel = listControl.ListViewModel;
+			int size = viewModel.Size;
+
+			for (int i = 0; i < size; i++)
+			{
+				var updateContainer = listControl.ContainerFromIndex(i) as TreeViewItem;
+				if (updateContainer != null)
+				{
+					if (isMultiSelect)
+					{
+						var targetNode = viewModel.GetNodeAt(i);
+						if (targetNode != null)
+						{
+							if (viewModel.IsNodeSelected(targetNode))
+							{
+								VisualStateManager.GoToState(updateContainer, "TreeViewMultiSelectEnabledSelected", false);
+							}
+							else
+							{
+								VisualStateManager.GoToState(updateContainer, "TreeViewMultiSelectEnabledUnselected", false);
+							}
+						}
+					}
+					else
+					{
+						VisualStateManager.GoToState(updateContainer, "TreeViewMultiSelectDisabled", false);
+					}
+				}
+			}
+		}
+
+		protected override void OnApplyTemplate()
+		{
+			ListControl = GetTemplateChild(c_listControlName);
+			if (ListControl != null)
+			{
+				var listPtr = m_listControl;
+				var viewModel = listPtr.ListViewModel;
+				if (m_rootNode == null)
+				{
+					m_rootNode = new TreeViewNode();
+				}
+
+				if (ItemsSource != null)
+				{
+					viewModel.IsContentMode = true;
+				}
+				viewModel.PrepareView(m_rootNode);
+				viewModel.SetOwningList(ListControl);
+				viewModel.NodeExpanding({ this, &TreeView::OnNodeExpanding });
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)